meta.go

```diff
@@ -59,9 +61,23 @@ func getMeta(key string, metaSpace string, metaFile string, output io.Writer, js
 		if err != nil {
 			return err
 		}
-		data, err := lastSuccessfulMetaRequest.FetchLastSuccessfulMeta(jobDescription)
-		if err != nil {
-			return err
+		var sb strings.Builder
+		var data []byte
+		externalMetaKey := jobDescription.MetaKey()
+		err = getMeta(externalMetaKey, metaSpace, kDefaultMetaFile, &sb, true, nil)
+		if err == nil && sb.String() != "null" {
+			logrus.Debugf("Found data in external meta key %s", externalMetaKey)
+			data = []byte(sb.String())
+		} else {
+			logrus.Debugf("Fetching metadata from %s", jobDescription.External())
+			data, err = lastSuccessfulMetaRequest.FetchLastSuccessfulMeta(jobDescription)
+			if err != nil {
+				return err
+			}
+			err = setMeta(externalMetaKey, string(data), metaSpace, kDefaultMetaFile, true)
```

Comment:
Do we also need to set this when read from file?  Will screwdriver launcher take care of setting this metadata for meta?

Reply:
Done with a refactor to make it slightly easier to write.